services/identity/service.go

```diff
@@ -281,33 +292,61 @@ func (s *Service) Propagate(msg network.Body) {
 
 // getIdentityStorage returns the corresponding IdentityStorage or nil
 // if none was found
-func (s *Service) getIdentityStorage(id ID) *storage {
+func (s *Service) getIdentityStorage(id ID) *Storage {
 	s.identitiesMutex.Lock()
 	defer s.identitiesMutex.Unlock()
-	is, ok := s.identities[string(id)]
+	is, ok := s.Identities[string(id)]
 	if !ok {
 		return nil
 	}
 	return is
 }
 
 // setIdentityStorage saves an IdentityStorage
-func (s *Service) setIdentityStorage(id ID, is *storage) {
+func (s *Service) setIdentityStorage(id ID, is *Storage) {
 	s.identitiesMutex.Lock()
 	defer s.identitiesMutex.Unlock()
 	log.Lvlf3("%s %x %v", s.Context.ServerIdentity(), id[0:8], is.Latest.Device)
-	s.identities[string(id)] = is
+	s.Identities[string(id)] = is
+}
+
+// saves the actual identity
+func (s *Service) save() {
```

Comment:
It seems to me that decision should be left to the user. Then in the place you use `save()`, because you are the user for the moment you can panic or whatever...